src/dds/rs-dds-internal-data.cpp

```diff
@@ -0,0 +1,178 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include "rs-dds-internal-data.h"
+
+#include <librealsense2/h/rs_sensor.h>
+#include <proc/processing-blocks-factory.h>
+#include <proc/color-formats-converter.h>
+#include <proc/depth-formats-converter.h>
+#include <proc/y8i-to-y8y8.h>
+#include <proc/y12i-to-y16y16.h>
+#include <proc/y12i-to-y16y16-mipi.h>
+
+namespace librealsense {
+
+static const std::string RS405_PID = "0B5B";
+static const std::string RS415_PID = "0AD3";
+static const std::string RS435_RGB_PID = "0B07";
+static const std::string RS435I_PID = "0B3A";
+static const std::string RS455_PID = "0B5C";
+static const std::string RS457_PID = "ABCD";
+
+
+std::vector< rs2_format > target_formats( rs2_format source_format )
+{
+    // Mapping from source color format to all of the compatible target color formats.
+    std::vector< rs2_format > formats = { RS2_FORMAT_RGB8, RS2_FORMAT_RGBA8, RS2_FORMAT_BGR8, RS2_FORMAT_BGRA8 };
+
+    switch( source_format )
+    {
+    case RS2_FORMAT_YUYV:
+        formats.push_back( RS2_FORMAT_YUYV );
+        formats.push_back( RS2_FORMAT_Y16 );
+        formats.push_back( RS2_FORMAT_Y8 );
+        break;
+    case RS2_FORMAT_UYVY:
+        formats.push_back( RS2_FORMAT_UYVY );
+        break;
+    default:
+        throw std::runtime_error( "Format is not supported for mapping" );
```

Comment:
Should this throw? Don't we want an identity somehow? E.g., what if the source is `rgb8`?

Reply:
This function is being called internally in this file only for YUYV and UYVY formats, to shorten the code line and make it more readable. For a format like `rgb8` no converter is needed so the sensor proxy will create an identity converter for it.

Reply:
I still think we shouldn't throw in this code, but -- knowing that this code will be removed in the PR -- let's move on...